tap_helpscout/streams/abstract.py

```diff
@@ -0,0 +1,220 @@
+from abc import abstractmethod, ABC
+from typing import Dict, Tuple, List, Iterator, Set
+from singer.metadata import get_standard_metadata, to_list, to_map, write
+from singer.bookmarks import ensure_bookmark_path
+from singer import metrics, Transformer, write_state
+import singer
+from tap_helpscout.transform import transform_json
+from tap_helpscout.helpers import parse_date
+
+logger = singer.get_logger()
+
+
+class BaseStream:
```

Comment:
We should make BaseStream an ABC so that all of the classes inheriting from BaseStream get it and don't need to be passed ABC repeatedly